src/Date.php

```diff
@@ -28,14 +28,14 @@ private function ensureValid(string $date)
             throw new \InvalidDateException();
        }
 
-        $this->date = $dt;
+        $this->date = $dt->format('Y-m-d');
```

Comment:
es macht meistens sinn, sich ein objekt (z.b. datetime) möglichst lange zu merken. du erstellst hier ein objekt und konvertierst es gleich wieder zu einem string (mittels format), was dir flexibilität nimmt. die formatierung des objektes zu einem string hätte ich in der toString Methode erwartet.